dmipy/utils/invariants.py

```diff
@@ -0,0 +1,576 @@
+import numpy as np
+from scipy.special import  lpmv, factorial
+from os.path import join as pjoin, dirname
+import pkg_resources
+from os.path import join
+from scipy.sparse import coo_matrix, load_npz
+from dipy.core.geometry import cart2sphere
+from dipy.reconst.shm import real_sym_sh_mrtrix
+from dipy.reconst.shm import real_sph_harm
+
+DATA_PATH = pkg_resources.resource_filename(
+    'dmipy', 'data'
+)
+
+def _get_index(l,m):
+    if l==0:
+        idx = 0
+    else:
+        i0 = ((l-1) * l)//2
+
+        idx = i0 + (m+l)
+
+    return int(idx)
+
+def _LegendreP(m,l,x):
+    return (-1)**m * lpmv(m,l,x)
+
+
+def sh_matrix(sh_order, vecs, sh_base = 'mrtrix'):
+    r""" Compute Spherical Harmonics (SH) matrix M
+    Parameters
+    ----------
+    sh_order : int, even
+        Truncation order of the SH.
+    vecs : array, Nx3
+        array unit directions [X,Y,Z].
+    sh_base : string,
+        the base of the SH.
+
+    Returns
+    --------
+    M : array,
+        The base of the SH sampled according to vecs.
+    """
+
+    r, theta, phi = cart2sphere(vecs[:, 0], vecs[:, 1], vecs[:, 2])
+    theta[np.isnan(theta)] = 0
+
+    n_c = (sh_order + 1) * (sh_order + 2) // 2
+
+    if sh_base == 'mrtrix':
+         M, m, l = real_sym_sh_mrtrix(sh_order, theta, phi)
+    else:
```

Comment:
That was my intention. That any unknown SH base will redirect to the gaunt one. But I can make mrtrix the default if you prefer

Reply:
It's not so much my preference what matters ;-)

I'm looking at it from an error-prone perspective. A person could give a misspelled name, e.g. 'Mrtrix' with a capital, and then get a gaunt basis without him knowing that he misspelled it. It would be very difficult to know what went wrong without more knowledge.

For this reason it's better to always throw an error in case of misspelling, and not default to a certain value.

Reply:
Ok, you are a wise man! :)

Reply:
There are three options, no? What about the gaunt basis?